File: server/functions/src/push-message.ts
/**
 * Class which represents the data of a pushmessage sent to the device
 */
export class PushMessage {
  content: string;
  datetime: string;
  host: string;

  /**
   *
   * @param {string} content of the QR-code
   * @param {string} datetime moment van scanner
   * @param {string} host van de website waar de QR-code is gescanned
   * @param {string} data encrypted data
   */
  constructor(content: string, datetime: string, host: string) {
    this.content = content;
    this.datetime = datetime;
    this.host = host;
  }
}

File: qrcode_receiver/lib/model/push_message.dart
class PushMessage{
  String host;
  DateTime datetime;
  String content;
  String data;

  PushMessage.fromJson(Map<String, dynamic> json){
    host = json['host'] ?? 'Unknown';
    datetime = DateTime.tryParse(json['datetime']);
    content = json['content'];
    data = json['data'];
  }
}
